Makefile

```diff
@@ -60,31 +60,67 @@ test-e2e: export OASIS_INDEXER_E2E = true
 test-e2e:
 	@$(GO) test -race -coverpkg=./... -coverprofile=coverage.txt -covermode=atomic -v ./tests/e2e
 
-fill-cache-for-e2e-regression: nexus
+fill-cache-for-e2e-regression:
+	@$(MAKE) fill-cache-for-e2e-regression-eden
+	@$(MAKE) fill-cache-for-e2e-regression-damask
+
+fill-cache-for-e2e-regression-eden: nexus
+	@./tests/e2e_regression/ensure_consistent_config.sh
+	./nexus --config tests/e2e_regression/eden/e2e_config_1.yml analyze
+	./nexus --config tests/e2e_regression/eden/e2e_config_2.yml analyze
+
+fill-cache-for-e2e-regression-damask: nexus
 	@./tests/e2e_regression/ensure_consistent_config.sh
-	./nexus --config tests/e2e_regression/e2e_config_1.yml analyze
-	./nexus --config tests/e2e_regression/e2e_config_2.yml analyze
+	./nexus --config tests/e2e_regression/damask/e2e_config_1.yml analyze
+	./nexus --config tests/e2e_regression/damask/e2e_config_1.yml analyze
 
 # Run the api tests locally, assuming the environment is set up with an oasis-node that is
 # accessible as specified in the config file.
-test-e2e-regression: nexus
+test-e2e-regression: test-e2e
+	@$(MAKE) test-e2e-regression-eden
+	@$(MAKE) test-e2e-regression-damask
+
+test-e2e-regression-eden: nexus
+	@./tests/e2e_regression/ensure_consistent_config.sh
+	./nexus --config tests/e2e_regression/eden/e2e_config_1.yml analyze
+	./nexus --config tests/e2e_regression/eden/e2e_config_2.yml analyze
+	@$(ECHO) "$(CYAN)*** Analyzers finished; starting api tests...$(OFF)"
+	./tests/e2e_regression/run.sh eden
+
+
+test-e2e-regression-damask: nexus
 	@./tests/e2e_regression/ensure_consistent_config.sh
-	./nexus --config tests/e2e_regression/e2e_config_1.yml analyze
-	./nexus --config tests/e2e_regression/e2e_config_2.yml analyze
+	./nexus --config tests/e2e_regression/damask/e2e_config_1.yml analyze
+	./nexus --config tests/e2e_regression/damask/e2e_config_2.yml analyze
 	@$(ECHO) "$(CYAN)*** Analyzers finished; starting api tests...$(OFF)"
-	./tests/e2e_regression/run.sh
+	./tests/e2e_regression/run.sh damask
 
 # Accept the outputs of the e2e tests as the new expected outputs.
 accept-e2e-regression:
+	@$(MAKE) accept-e2e-regression-eden
+	@$(MAKE) accept-e2e-regression-damask
+
+accept-e2e-regression-eden:
+	@# Delete all old expected files first, in case any test cases were renamed or removed.
+	rm -rf ./tests/e2e_regression/eden/expected
+	@# Copy the actual outputs to the expected outputs.
+	cp -r  ./tests/e2e_regression/eden/{actual,expected}
+	@# The result of the "spec" test is a special case. It should always match the
+	@# current openAPI spec file, so we symlink it to avoid having to update the expected
+	@# output every time the spec changes.
+	rm ./tests/e2e_regression/eden/expected/spec.body
+	ln -s  ../../../../api/spec/v1.yaml ./tests/e2e_regression/eden/expected/spec.body
```

Comment:
Hmm I'd prefer not to split the logic into multiple files, esp if we end up doing the same for `fill-cache` and `test-e2e`.

Makefile doesn't pass the shell vars to the subcommands though. I'll keep poking at it

Reply:
Makefile (or rather `make`) doesn't technically pass env variables to bash; it substitutes them into the command _before_ the command reaches bash. Note that my snippet above has `$(EDEN)`, not `${EDEN}`. The former is `make` syntax for var substitution. Does that work? I have toy examples locally that work but I might be missing something.

Reply:
Hm I think the main issue I'm seeing rn is that the makefile `ifndef` needs to be at the top level and [not within a recipe](https://stackoverflow.com/questions/17620887/makefile-conditional-error). To check whether a variable is defined within a recipe, we'd use the shell scripting, but bash and zsh have different syntax for doing this and having conditional logic based on the shell in a makefile seems to be a [cardinal sin](https://www.gnu.org/software/make/manual/html_node/Environment.html). (Side note: Recent versions of bash/zsh both use the `[[ -v SUITE ]]` syntax, but it isn't supported in the default bash version on macs (3.2))

Reply:
Well we can just unindent the `ifndef` block and it works.. though it looks messy:
```
accept-e2e-regression-suite:
ifndef SUITE
	$(error SUITE is undefined)
endif
	@# Delete all old expected files first, in case any test cases were renamed or removed.
	rm -rf ./tests/e2e_regression/$(SUITE)/expected
```

We could instead define a separate recipe to check that the env var is set: 
```
check-suite:
ifndef SUITE
	$(error SUITE is undefined)
endif
```

Reply:
Huh I've certainly learned more about Makefiles through this :). Sorry it's such a time-consuming process. Nice solution in the end IMO!

> though it looks messy

Fun fact: `make` is from the 70s. It's amazing it won't die.
The syntax is about as charming as the tan-striped nicotine-stained couches of the 70s ...